Dockerfile

```diff
@@ -85,6 +85,9 @@ RUN curl -o /usr/local/bin/ecs-cli https://s3.amazonaws.com/amazon-ecs-cli/ecs-c
     echo "$(curl -s https://s3.amazonaws.com/amazon-ecs-cli/ecs-cli-linux-amd64-latest.md5) /usr/local/bin/ecs-cli" | md5sum -c - && \
     chmod +x /usr/local/bin/ecs-cli
 
+# Install Helm
+RUN curl https://raw.githubusercontent.com/helm/helm/master/scripts/get-helm-3 | DESIRED_VERSION=v3.0.0 bash
```

Comment:
This should be done a little differently:
1. Use a tag instead of a branch, so that we know what binary is being installed.
1. Grab the binary directly and install it in /usr/local/bin/ instead of running an unaudited shell script.
1. Find a hash from upstream's releases and check that the download matches the hash.  The hash should be checked into VCS so builds will fail on a bad/malicious download

Reply:
I will pin a checksum and move the install logic to our side. But I will complain a bit first for my own benefit. :) 

The script is maintained and published by the same trusted party publishing the binary. You are either trusting the party or not. Accepting less code from them and bringing on your plate (3 lines it may be) is a bit much, no? Although the code is running at a more sensitive time...

If security is a big concern here perhaps this should be more of a base image that each team extends? I think it may even be easier to maintain given it allows us to keep team boundaries. Although, you wouldn't be there to review :)

Anyhow, I really do appreciate the thoughtful review. Thanks, Ross!

Reply:
It's not quite that I don't trust the software from the helm developers.  There are two reasons for my ask:

First: we need to install a known version of software that doesn't change.  If we install from a branch, the tool will change in unexpected ways underneath of users - and helm is very picky about always using the same version.

Second: When downloading with curl, nothing ensures that we actually downloaded the thing we wanted.  To check that, we need to validate the download against a known good checksum or signature - and that means not downloading the checksum during the automated process.  (If something is compromised to force a download of bad binaries, then downloading similarly bad checksums is a risk too.)

This is why most of the software is installed via a package distribution system that we trust (apt, docker, etc).  But whenever possible, we should validate all other downloads.  This isn't possible in all cases (AWS doesn't publish checksums/signatures for ecs-cli).  But all github releases should be published with checksums.

And if you'd rather use this as a base image and extend as you prefer, that's okay too.  I still think you should be concerned about tool stability and supply chain security - but you don't work for me so I can't make you :)